pkg/vmprovider/providers/vsphere/virtualmachine/devices.go

```diff
@@ -32,42 +33,43 @@ func CreatePCIDevices(
 	pciDevicesFromVMClass v1alpha1.VirtualDevices,
```

Comment:
Can we handle this higher in the call stack? IMO this change is poor ergonomics here because the caller still needs to provide both arguments but only one will ever be used. In the earlier class ConfigSpec changes I worked to @sreyasn to do these FFS checks early on so most things can just depend on nil vs non-nil pointers & empty vs nonempty collections.

If we can't do that, maybe time to split this function into two?

Reply:
There's a push/pull between re-using code and FSS split-logic. I tend to err on the side of keeping the FSS check as early as possible to prevent things like possible NPEs in the call stack.

Reply:
Fixed.  Moved the FSS check up the call stack.